ex5/dist/main.js

```diff
@@ -0,0 +1,156 @@
+class Main {
+	constructor() {
+		this.itemClient = new ItemClient();
+	}
+
+	init = async () => {
+		const addItemButton = document.getElementById('list-item-submit');
+		addItemButton.addEventListener('click', this.handleItem);
+
+		const deleteAllBtn = document.getElementById('list-item-clear');
+		deleteAllBtn.addEventListener('click', this.handleDeleteAll);
+
+		await this.renderItems();
```

Comment:
all of the DOM manipulation is not really asynchronous
you could simplify the flow a lot by making everything in the client synchronous (apart from the server fetch commands - these really are async)